### PR TITLE
Use one lean cli root folder per organization

### PR DESCRIPTION
**Pull request by jhonabreul**
This changes will make the CLI use the root `lean init` folder for a single organization:

- The user will have to run `lean init` in a single directory for each organization they belong to (and want to use the CLI with):
    - On init, the user has to give the ID or name of the organization to use (or select it interactively).
- The organization ID is stored in the `lean.json` file and it will be used by every command that needs it (using the `OrganizationManager`, if possible).
- Users using a previous CLI version (no `organization-id` in the `lean.json` file) will get a message informing them of the changes and what to do to keep using the CLI.

**Note:** the check for existence of the organization id the lean config file on every command adds a small overhad of around 6ms, which is acceptable considering that it is expected to be removed in the near future.

### CODE REVIEW

File: lean/commands/data/download.py [hunk 1]
```diff
@@ -302,30 +286,16 @@ def _confirm_payment(organization: QCFullOrganization, products: List[Product])
     confirm("Continue?", abort=True)
 
 
-def _get_organization_by_name_or_id(user_input: str) -> QCFullOrganization:
-    """Finds an organization by name or id.
+def _get_organization() -> QCFullOrganization:
+    """Gets the working organization/
```
Comment (Martin-Molinero) [inline lean/commands/data/download.py:290]: typo `/` ?

File: lean/commands/lean.py [hunk 2]
```diff
@@ -15,12 +15,32 @@
...
+    except MoreInfoError:
+        # There is no config, proceed in case `lean init` is being invoked
+        return
+
+    # `lean init` was already called. Let's check whether this is an old Lean CLI directory
+    organization_manager = container.organization_manager
+    if organization_manager.get_working_organization_id() is None:
+        raise RuntimeError(
+            "This is an old Lean CLI root folder.\n"
+            "From now on, a Lean CLI root folder must be created for each organization for improved organization.\n"
```
Comment (Martin-Molinero) [inline lean/commands/lean.py:45]: nit - `organization for improved organization`  maybe we can rephrase this one

File: lean/components/docker/lean_runner.py [hunk 3]
```diff
@@ -183,7 +183,7 @@ def get_basic_docker_config(self,
         # Install the required modules when they're needed
         if lean_config.get("data-provider", None) == "QuantConnect.Lean.Engine.DataFeeds.DownloaderDataProvider" \
             and lean_config.get("data-downloader", None) == "TerminalLinkDataDownloader":
-            self._module_manager.install_module(TERMINAL_LINK_PRODUCT_ID, lean_config["job-organization-id"])
+            self._module_manager.install_module(TERMINAL_LINK_PRODUCT_ID, lean_config["organization-id"])
```
Comment (Martin-Molinero) [inline lean/components/docker/lean_runner.py:186]: `job-organization-id` is what lean config.json is using, see https://github.com/QuantConnect/Lean/blob/master/Launcher/config.json#L71 , which is != than what the lean cli lean.json has, which will now be `organization-id` ?

File: lean/components/util/organization_manager.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    def try_get_working_organization_id(self) -> Optional[str]:
+        """Gets the id of the working organization in the current Lean CLI directory.
+
+        :return: the id of the working organization
+        :raises RuntimeError: if the working organization is not found in the lean config
+        """
+        organization_id = self.get_working_organization_id()
+
+        if organization_id is None:
+            raise RuntimeError("The working organization for this Lean CLI folder could not be determined.")
```
Comment (Martin-Molinero) [inline lean/components/util/organization_manager.py:55]: Maybe we can add that the user should run `lean init`?

File: lean/models/click_options.py [hunk 5]
```diff
@@ -82,7 +82,7 @@ def get_options_attributes(configuration: Configuration, default_lean_config_key
 
 
 def get_default_key(configuration: Configuration):
-    return "job-organization-id" if configuration.is_type_organization_id else configuration._id
+    return configuration._id
```
Comment (Martin-Molinero) [inline lean/models/click_options.py:85]: hm tricky related to the other comment above I think, `job-organization-id` is what lean is expecting

File: lean/models/lean_config_configurer.py [hunk 6]
```diff
@@ -58,8 +58,6 @@ def configure_credentials(self, lean_config: Dict[str, Any]) -> None:
         """Configures the credentials in the Lean config for this brokerage and saves them persistently to disk.
         :param lean_config: the Lean configuration dict to write to
         """
-        if self._installs:
-            lean_config["job-organization-id"] = self.get_organzation_id()
```
Comment (Martin-Molinero) [inline lean/models/lean_config_configurer.py:62]: Same comment about `job-organization-id`

File: tests/commands/test_live.py [hunk 7]
```diff
@@ -660,7 +651,7 @@ def test_live_non_interactive_falls_back_to_lean_config_for_multiple_data_feed_s
                 file.write(json.dumps({
                     **missing_options_config,
                     "data-folder": "data",
-                    "job-organization-id": "abc"
```
Comment (Martin-Molinero) [inline tests/commands/test_live.py:663]: Same comment about `job-organization-id`

**Review by Martin-Molinero — COMMENTED** [review]
Cool! Leaving minor comments

File: lean/components/docker/lean_runner.py (hunk 3 above)
Comment (jhonabreul) [reply lean/components/docker/lean_runner.py:186]: Right, fixing

File: lean/models/click_options.py (hunk 5 above)
Comment (jhonabreul) [reply lean/models/click_options.py:85]: This one is not used anymore, there are no `OrganizationIdConfiguration` configurations anymore.

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**